mesheryctl/internal/cli/root/root.go

```diff
@@ -141,33 +141,42 @@ func initConfig() {
 	//  perf:
 	//	  authTokenURI:    "/gettoken",
 	//	  loadTestSmpURI: "/perf/load-test-smp",
+
+	defaultsMap := map[string]interface{}{
+		"baseMesheryURL": "http://localhost:9081/api",
+		"perf": map[string]interface{}{
+			"authTokenURI":   "/gettoken",
+			"loadTestSmpURI": "/perf/load-test-smp]",
+			"loadTestURI":    "/perf/load-test",
+		},
+		"ctlversion": map[string]interface{}{
+			"build":     version,
+			"commitsha": commitsha,
+		}}
+
 	if mctlCfgFile != "" {
 		viper.SetConfigFile(mctlCfgFile)
 		if err := viper.ReadInConfig(); err == nil {
 			log.Debugf("Using mesheryctl config file: %s", viper.ConfigFileUsed())
 		} else {
-			log.Fatal(err)
+			log.Info(err)
+			generateMesheryctlConfigUsingDefaultValues(defaultsMap)
 		}
 	} else {
-		setMesheryctlConfigDefaults(map[string]interface{}{
-			"baseMesheryURL": "http://localhost:9081/api",
-			"perf": map[string]interface{}{
-				"authTokenURI":   "/gettoken",
-				"loadTestSmpURI": "/perf/load-test-smp]",
-				"loadTestURI":    "/perf/load-test",
-			},
-			"ctlversion": map[string]interface{}{
-				"build":     version,
-				"commitsha": commitsha,
-			},
-		})
+		generateMesheryctlConfigUsingDefaultValues(defaultsMap)
 	}
 }
 
-// setMesheryctlConfigDefaults loads the hardcoded defaults in to viper kv store
-func setMesheryctlConfigDefaults(defaults map[string]interface{}) {
+// generateMesheryctlConfigUsingDefaultValues creates config file using the hardcoded defaults in viper kv store
+func generateMesheryctlConfigUsingDefaultValues(defaults map[string]interface{}) {
+	log.Info("Creating and Using defaults config for mesheryctl")
 	v := viper.GetViper()
 	for key, value := range defaults {
 		v.SetDefault(key, value)
 	}
+	err := v.WriteConfigAs(utils.MesheryFolder + "/mesheryctlConfig.yaml")
```

Comment:
Would we have the permissions to write. I am assuming yes

Reply:
Yes we would